roles/qemu_setup/tasks/main.yml

```diff
@@ -17,13 +17,47 @@
       - bridge-utils
       - cloud-image-utils
       - cpu-checker
+      - guestfs-tools
+      - libnss-libvirt
       - libvirt-clients
       - libvirt-daemon-system
       - qemu-kvm
       - virt-manager
       - virtinst
   become: true
 
+- name: Add user to virt groups
+  ansible.builtin.user:
+    name: "{{ username }}"
+    groups: kvm, libvirt
+    append: true
+  become: true
+
+- name: Enable access to images folder in AppArmour
+  ansible.builtin.lineinfile:
+    path: /etc/apparmor.d/abstractions/libvirt-qemu
+    line: "  /var/lib/libvirt/images/* rwk,"
+  become: true
+
+- name: Check for libvirt_guest in nsswitch.conf
+  ansible.builtin.lineinfile:
+    path: /etc/nsswitch.conf
+    regex: ^hosts:.*libvirt_guest*
+    state: absent
+  check_mode: true
+  changed_when: false
+  register: libvirt_guest
+  become: false
+
+- name: Enable libvirt_guest dns via nss
+  ansible.builtin.lineinfile:
+    path: /etc/nsswitch.conf
+    regex: ^(hosts:.*)
+    line: \1 libvirt_guest
+    backrefs: true
+  become: true
+  when: not libvirt_guest.found
```

Comment:
I haven't tried this myself, but I feel that this and the previous step could be done in a single step. Might need a bit of playing with the regexes.

Reply:
I looked at that @iomartin and it is not easy and makes the code quite hard to follow. I think this is a cleaner approach.